File: src/test/java/com/koddy/server/common/config/TestAopConfig.java
package com.koddy.server.common.config;

import com.koddy.server.global.aop.AccessControlAop;
import com.koddy.server.global.log.LoggingStatusManager;
import com.koddy.server.global.log.LoggingTracer;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.EnableAspectJAutoProxy;

@TestConfiguration
@EnableAspectJAutoProxy
public class TestAopConfig {
    @Bean
    public AccessControlAop accessControlAop() {
        return new AccessControlAop();
    }

    @Bean
    public LoggingStatusManager loggingStatusManager() {
        return new LoggingStatusManager();
    }

    @Bean
    public LoggingTracer loggingTracer() {
        return new LoggingTracer(loggingStatusManager());
    }
}

File: src/test/kotlin/com/koddy/server/common/config/TestAopConfig.kt
package com.koddy.server.common.config

import com.koddy.server.global.aop.AccessControlAop
import com.koddy.server.global.log.LoggingStatusManager
import com.koddy.server.global.log.LoggingTracer
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.EnableAspectJAutoProxy

@TestConfiguration
@EnableAspectJAutoProxy
class TestAopConfig {
    @Bean
    fun accessControlAop(): AccessControlAop = AccessControlAop()

    @Bean
    fun loggingStatusManager(): LoggingStatusManager = LoggingStatusManager()

    @Bean
    fun loggingTracer(): LoggingTracer = LoggingTracer(loggingStatusManager())
}
